Code/Framework/AtomCore/Tests/InstanceDatabase.cpp

```diff
@@ -225,23 +253,51 @@ namespace UnitTest
         for (size_t i = 0; i < threadCountMax; ++i)
         {
             threads.emplace_back(
-                [&instanceManager, &threadCount, &cv, &guids, &assets, &durationSeconds]()
+                [&instanceManager, &threadCount, &cv, &guids, &instances, &assets, &durationSeconds, &deferRemoval, &orphan, &referenceTableMutex]()
                 {
                     AZ::Debug::Timer timer;
                     timer.Stamp();
 
+                    // If both deferRemoval and orphan are true, use 3 as the modulus so that each iteration will either orphan, release, or create
+                    // If only one is true, use 2 as the modulus so each iteration will create or defer/orphan
+                    // If neither are used, the modulus will be ignored so it doesn't matter
+                    int randomModulus = deferRemoval && orphan ? 3 : 2;
```

Comment:
I found this code to be a bit hard to reason about. Here's a suggested alternative, I'll leave it up to you...
```
bool doCreate = false;
bool doOrphan = false;
bool doDeferredRelease = false;

if(deferRemoval && orphan)
{
    switch(rand() % 3)
    {
        case 0: doCreate = true; break;
        case 1: doOrphan = true; break;
        case 2: doDeferredRelease = true; break;
    }
}
else if(deferRemoval)
{
    switch(rand() % 2)
    {
        case 0: doCreate = true; break;
        case 1: doDeferredRelease = true; break;
    }
}
else if(orphan)
{
    switch(rand() % 2)
    {
        case 0: doCreate = true; break;
        case 1: doOrphan = true; break;
    }
}
else
{
    doCreate = true;
}

if(doOrphan)
{
    instanceManager.TEMPOrphan(instanceId);
}
if(doDeferredRelease)
{
    referenceTableMutex.lock();
    instances[index] = instance;
    referenceTableMutex.unlock();
}
if(doCreate)
{
    Instance<TestInstanceA> instance = instanceManager.FindOrCreate(instanceId, assets[index]);
    ...

    if(deferRemoval)
    {                              
        referenceTableMutex.lock();
        instances[index] = instance;
        referenceTableMutex.unlock();
    }
}
```

Reply:
That is definitely more readable. I will make that (or similar) change.